cpu/cpu.go

```diff
@@ -146,12 +146,20 @@ func (p *Plugin) GetMetricTypes(cfg plugin.ConfigType) ([]plugin.MetricType, err
 // It returns error in case retrieval was not successful
 func (p *Plugin) CollectMetrics(metricTypes []plugin.MetricType) ([]plugin.MetricType, error) {
 	metrics := []plugin.MetricType{}
-	for _, metricType := range metricTypes {
-		if !p.initialized {
-			if err := p.init(metricType.Config().Table()); err != nil {
-				return nil, err
-			}
+	if len(metricTypes) == 0 {
+		return nil, nil
+	}
+	if !p.initialized {
+		if err := p.init(metricTypes[0].Config().Table()); err != nil {
+			return nil, err
 		}
+	} else {
```

Comment:
@obourdon Sorry for not being precise. I forgot to add that if plugin is not initiliazed, `p.init` is called. Inside `p.init` apart from reading configuration, there is a [call](https://github.com/obourdon/snap-plugin-collector-cpu/blob/e59905586ee357391e1eecab6f7e1f0178038b50/cpu/cpu.go#L226) to `getStats`. I think `p.init` should not call `getStats`, should only initialize state of the plugin, so later on `getStats` can be called safely.

I suggest to remove `getStats` from `p.init`, and then above block can be rewritten to

```go
if !p.initialized {
	if err := p.init(metricTypes[0].Config().Table()); err != nil {
		return nil, err
	}
}

if err := getStats(p.stats, p.prevMetricsSum, p.cpuMetricsNumber, p.snapMetricsNames, p.procStatMetricsNames); err != nil {
	return nil, err
 }
```